SL/DATEV.pm

```diff
@@ -1080,21 +1080,35 @@ sub check_document_export {
 
   return 1;
 
-  # TODO maybe needed
+}
+
+sub check_all_bookings_have_documents {
+  my $self   = shift;
+  my %params = @_;
+
+  die "Need from date" unless $params{from};
+  die "Need to date"   unless $params{to};
+
+  $self->from($params{from});
+  $self->to($params{to});
+
+  my $fromto = $self->fromto;
   # not all last month ar ap gl booking have an entry -> rent ?
-  my $query = <<"SQL";
+  my $query = qq|
   select distinct trans_id,object_id from acc_trans
   left join files on files.object_id=trans_id
-  where date_trunc('month', transdate) = date_trunc('month', current_date - interval '1 month')
+  where $fromto
```

Comment:
Wäre es nicht sicherer, dass auf parametrisierte Query umzuschreiben? Oder geht das hier nicht?

Reply:
Hab ich auch überlegt, der code wird aber schon woanders verwendet und wollte jetzt kein tieferes Refactoring machen.
Ich hab deswegen minimal die Klassenfunktion auf korrekte Eingabewerte überprüft, das passiert in f3c5d344a9